src/lib/score/graphics/widgets/QGraphicsMultiSliderXY.cpp

```diff
@@ -0,0 +1,164 @@
+#include <score/graphics/widgets/QGraphicsMultiSliderXY.hpp>
+#include <score/model/Skin.hpp>
+#include <score/serialization/StringConstants.hpp>
+#include <score/tools/Debug.hpp>
+
+#include <ossia/detail/math.hpp>
+#include <ossia/network/domain/domain_functions.hpp>
+#include <ossia/network/value/value_conversion.hpp>
+
+#include <QGraphicsSceneMouseEvent>
+#include <QPainter>
+
+#include <wobjectimpl.h>
+W_OBJECT_IMPL(score::QGraphicsMultiSliderXY);
+
+namespace score
+{
+score::QGraphicsMultiSliderXY::QGraphicsMultiSliderXY(QGraphicsItem* parent)
+{
+  auto& skin = score::Skin::instance();
+  setCursor(skin.CursorPointingHand);
+  this->setAcceptedMouseButtons(Qt::LeftButton | Qt::RightButton);
+}
+
+void score::QGraphicsMultiSliderXY::setPoint(const QPointF& r)
+{
+  SCORE_TODO;
+}
+
+void score::QGraphicsMultiSliderXY::paint(
+    QPainter* painter, const QStyleOptionGraphicsItem* option, QWidget* widget)
+{
+
+  tab = m_value.get<std::vector<ossia::value>>();
+
+
+  // Draw the background
+  painter->setBrush(QColor(qRgba(10, 10, 10, 255)));
+  painter->fillRect(QRectF(0, 0, width(), height()), QColor(qRgba(10, 10, 10, 255)));
+
+  // Draw cursors
+  for (ossia::value& v : tab)
+  {
+
+    painter->fillRect(
+        QRectF((v.get<ossia::vec2f>()[0] - cursorSize.x / 2) * width(),
+               (v.get<ossia::vec2f>()[1] - cursorSize.y / 2) * height(),
+               cursorSize.x * width(),
+               cursorSize.y * height()),
+        QColor(qRgba(0, 0, 255, 255)));
+  }
+}
+
+ossia::value score::QGraphicsMultiSliderXY::value() const
+{
+  return m_value;
+}
+
+void score::QGraphicsMultiSliderXY::setRange(const ossia::value& min, const ossia::value& max)
+{
+  this->min = ossia::convert<float>(min);
+  this->max = ossia::convert<float>(max);
+}
+
+void score::QGraphicsMultiSliderXY::setRange(const ossia::domain& dom)
+{
+  auto [min, max] = ossia::get_float_minmax(dom);
+
+  if (min)
+    this->min = *min;
+  if (max)
+    this->max = *max;
+}
+
+void score::QGraphicsMultiSliderXY::setValue(ossia::value v)
+{
+  prepareGeometryChange();
+  m_value = v;
+  update();
+}
+
+void score::QGraphicsMultiSliderXY::mouseMoveEvent(QGraphicsSceneMouseEvent* event)
+{
+  if (m_grab) // The cursor’s position is updated as the mouse moves, constrained within the widget’s bounds.
+  {
+    tab[selectedCursor] = ossia::vec2f{(float)std::clamp((event->pos().x() / width()), 0., 1.),
+                                       (float)std::clamp((event->pos().y() / height()), 0., 1.)};
+    m_value = std::vector<ossia::value>(tab.begin(), tab.end());
+    sliderMoved();
+    update();
+  }
+}
+
+void score::QGraphicsMultiSliderXY::mousePressEvent(QGraphicsSceneMouseEvent* event)
+{
+  float x = event->pos().x();
+  float y = event->pos().y();
+
+  // If the left mouse button is pressed on an existing cursor, that cursor can be moved.
+  if (event->button() & Qt::LeftButton)
+  {
+    for (int v = 0; v < tab.size(); v++)
```

Comment:
std::ssize(tab) instead of tab.size() otherwise we get unsigned / signed comparison warnings